code/game/objects/effects/effect_system/effects_foam.dm

```diff
@@ -54,18 +54,17 @@
 		kill_foam()
 		return
 
-	var/fraction = 1/initial(lifetime)
```

Comment:
reagent duplication meme is a non-issue and quite frankly is why we had really shitty foam/smoke for so long